src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala

```diff
@@ -35,11 +38,14 @@ object CallMetadataBuilder {
   private type ExecutionMapTransformer = ExecutionMap => ExecutionMap
 
   object BackendValues {
-    def extract(job: Any): BackendValues = {
-      job match {
-        case ji: LocalJob => BackendValues("Local")
-        case ji: JesJob => BackendValues("JES", jobId = ji.jesRunId, status = ji.jesStatus)
-        case ji: SgeJob => BackendValues("SGE", jobId = ji.sgeJobNumber map { _.toString })
+    // FIXME needs to traverse the list of pluggable backends rather than hardcoding a list.
+    def extract(infos: ExecutionInfosByExecution): BackendValues = {
+      def extractValue(key: String): Option[String] = infos.executionInfos collectFirst { case i if i.key == key => i.value } flatten
+
+      infos.execution.backendType match {
+        case "Local" => BackendValues("Local", jobId = extractValue(LocalBackend.InfoKeys.Pid))
+        case "JES" => BackendValues("JES", jobId = extractValue(JesBackend.InfoKeys.JesRunId), status = extractValue(JesBackend.InfoKeys.JesStatus))
+        case "SGE" => BackendValues("SGE", jobId = extractValue(SgeBackend.InfoKeys.JobNumber))
```

Comment:
what if the backend was a composition? e.g. we trying to run a job on a distributed environment (Broad-clsters and GC and AWS) so we design a backend that is made of e.g. JES+SGE only for the sake of using the one that offers the best $$ for the task, should Cromwell be aware of which backend instance if the composite picking?. if we just had some form of:
BackendValues(someInstanceKey, jobId = extractValue(Backend.InfoKeys.JobNumber))
we could use any weird design of the backends tree and cromwell will still run